ml-proto/src/spec/check.ml

```diff
@@ -179,14 +179,14 @@ let rec check_expr c ts e =
 
   | SetLocal (x, e1) ->
     check_expr c [local c x] e1;
-    check_type [] ts e.at
+    check_type [local c x] ts e.at
```

Comment:
Here and below, it might be nice to factor out via `let t1 = local c x in` to avoid duplicating the expression.